frontend/packages/core/src/Contexts/workflow-storage-context/reducer.tsx

```diff
@@ -0,0 +1,74 @@
+import { removeLocalData, rotateDataFromAPI, storeLocalData } from "./helpers";
+import type { Action, BackgroundPayload, UserPayload, WorkflowStorageState } from "./types";
+
+/**
+ * Reducer for the WorkflowStorageContext
+ * This will act on the WorkflowStorageState and add / remove items from the temporary storage
+ * as well as localStorage, this will (optionally) keep all storage actions in one location for all
+ * workflows and allow for easier state hydration
+ */
+const workflowStorageContextReducer = (
+  state: WorkflowStorageState,
+  action: Action
+): WorkflowStorageState => {
+  switch (action.type) {
+    // Will add data to our temporary storage as well as the local storage
+    case "STORE_DATA": {
+      const { componentName, key, data, localStorage = true } = action.payload as UserPayload;
```

Comment:
UserPayload is a bit confusing here, isn't this the component payload?

Reply:
Yeah that's a better name, let me update